src/lib/y2partitioner/widgets/rescan_devices_button.rb

```diff
@@ -0,0 +1,53 @@
+require "yast"
+require "cwm/widget"
+require "y2partitioner/device_graphs"
+require "y2storage/storage_manager"
+
+Yast.import "Popup"
+
+module Y2Partitioner
+  module Widgets
+    # Button for rescanning system devices
+    class RescanDevicesButton < CWM::PushButton
+      def initialize
+        textdomain "storage"
+      end
+
+      def label
+        _("Rescan Devices")
+      end
+
+      # Shows a confirm message before reprobing
+      def handle
+        return nil unless continue?
+
+        reprobe
+        :redraw
+      end
+
+    private
+
+      def continue?
+        Yast::Popup.YesNo(
+          # TRANSLATORS
+          format(
+            _("Current changes will be discarted, including changes done by the proposal.\n" \
+              "Do you want to continue?")
```

Comment:
Not bad. I just wonder, wouldn't be good to be more explicit/scary? Another suggestion

In installation mode:

_Re-scanning the storage devices will invalidate all the configuration options set in the installer regarding storage, with no possibility to withdraw._

_That includes the result and settings of the guided setup as well as the manual changes performed in the expert partitioner._

In normal mode:
_Re-scanning the storage devices will invalidate all the previous changes with no possibility to withdraw._